content/blog/aws-lambda-efs/index.md

```diff
@@ -0,0 +1,228 @@
+
+```ts
+const filesystem = new aws.efs.FileSystem("filesystem");
+    const targets = [];
+    for (let i = 0; i < subnetIds.length; i++) {
+        targets.push(new aws.efs.MountTarget(`fs-mount-${i}`, {
+            fileSystemId: filesystem.id,
+            subnetId: subnetIds[i],
+            securityGroups: [vpc.vpc.defaultSecurityGroupId],
+        }));
+    }
+```
+
+To use EFS from our Lambda Function, we need to tell Lambda how to mount the NFS volume.  Specifically, what Unix user and group to use for access, and what folder and permissions to use as the root folder of the mount.  This is done using an EFS AccessPoint.
+
+```ts
+    const ap = new aws.efs.AccessPoint("ap", {
+        fileSystemId: filesystem.id,
+        posixUser: { uid: 1000, gid: 1000 },
+        rootDirectory: { path: "/www", creationInfo: { ownerGid: 1000, ownerUid: 1000, permissions: "755" } },
+    }, { dependsOn: targets });
+```
+
+Finally, each Lambda Function we want to run will need to be in our VPC, use our default security group, and mount the EFS file system via our AccessPoint.  We can write a small function that creates Lambda Functions with this configuration.  We can specify where we want to mount this into the Lambda Function - though we must mount it underneath `/mnt/`.
+
+```ts
+    function efsvpcCallback(name: string, f: aws.lambda.Callback<awsx.apigateway.Request, awsx.apigateway.Response>) {
+        return new aws.lambda.CallbackFunction(name, {
+            policies: [aws.iam.ManagedPolicies.AWSLambdaVPCAccessExecutionRole, aws.iam.ManagedPolicies.AWSLambdaFullAccess],
+            vpcConfig: {
+                subnetIds: vpc.privateSubnetIds,
+                securityGroupIds: [vpc.vpc.defaultSecurityGroupId],
+            },
+            fileSystemConfigs: [{ arn: ap.arn, localMountPath: "/mnt/storage" }],
+            callback: f,
+        });
+    }
+```
+
+We now have our infrastructure all set up to start using Lambda with EFS.  Let’s build an app!
+
+## Interact with EFS from an HTTP API
+
+For a first simple Lambda application, we'll expose an HTTP API that GETs files from, and POSTs files to, our EFS file system.  We'll implement this with API Gateway and two routes.  Using the helper we defined above, each of these is just a few lines of code.
+
+Note that we can use our normal Node.js `fs.readFileSync` and `fs.writeFileSync` APIs to interact with the EFS file system - no need for an AWS SDK to talk to EFS. This makes it easy to take existing software and libraries that can interact with the filesystem and run them inside Lambda.
+
+> Note: Yes, exposing this to the internet is insecure - it is recommended that you [add authentication or use a private endpoint](https://www.pulumi.com/docs/guides/crosswalk/aws/api-gateway/#controlling-and-managing-access-to-apis) before running this in production!
```

Comment:
```suggestion
> Note: Yes, exposing this to the internet is insecure - it is strongly recommended that you [add authentication or use a private endpoint](https://www.pulumi.com/docs/guides/crosswalk/aws/api-gateway/#controlling-and-managing-access-to-apis) before running this in production!
```

Reply:
I really, really don't think we should post insecure examples -- ever. It's way, way too common for folks to cut/paste things they see on our blogs.